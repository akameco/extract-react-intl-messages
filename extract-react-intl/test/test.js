const m = require('..')

const pattern = 'extract-react-intl/test/fixtures/**/*.js'
const locales = ['en', 'ja']

test('extract from file', async () => {
  process.env.BABEL_ENV = 'react-intl'
  const x = await m(locales, pattern, {
    cwd: `${__dirname}/fixtures`,
    extractFromFormatMessageCall: true
  })
  expect(x).toMatchSnapshot()
})

// TODO: fix
test.skip('babelrc path resolution', async () => {
  const x = await m(['en'], './extract-react-intl/test/resolution/**/*.js', {
    cwd: `${__dirname}/resolution`
  })
  expect(x).toMatchSnapshot()
})

test('babel plugin execution order', async () => {
  const x = await m(
    ['en'],
    './extract-react-intl/test/pluginOrdering/**/*.js',
    { cwd: `${__dirname}/pluginOrdering` }
  )
  expect(x).toMatchSnapshot()
})

test('error', async () => {
  expect.assertions(1)
  await m(locales, 'notfound', {
    cwd: `${__dirname}/fixtures`
  }).catch(error => {
    expect(error.message).toMatch('File not found')
  })
})
