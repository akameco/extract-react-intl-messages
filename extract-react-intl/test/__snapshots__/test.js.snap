// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin execution order 1`] = `
Object {
  "en": Object {
    "extract-react-intl.test.pluginOrdering.test": "auto",
  },
}
`;

exports[`babelrc path resolution 1`] = `
Object {
  "en": Object {
    "test": "test",
  },
}
`;

exports[`extract from file 1`] = `
Object {
  "en": Object {
    "components.App.1248161314": "Submit button",
    "components.App.hello": "hello",
    "components.App.world": "world",
    "components/Greeting/welcome": "Welcome {name}, you have received {unreadCount, plural, =0 {no new messages} one {{formattedUnreadCount} new message} other {{formattedUnreadCount} new messages}} since {formattedLastLoginTime}.",
  },
  "ja": Object {
    "components.App.1248161314": "",
    "components.App.hello": "",
    "components.App.world": "",
    "components/Greeting/welcome": "",
  },
}
`;
